diff --git a/config.sub b/config.sub
index dba16e8..5e19f90 100755
--- a/config.sub
+++ b/config.sub
@@ -1756,6 +1756,9 @@ case $os in
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
 	     | fiwix* )
 		;;
+	# xPack GNU RISC-V Embed GCC uses riscv-none-embed-*.
+	embed)
+		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
 		# Don't forget version if it is 3.2v4 or newer.
diff --git a/gdb/python/python-config.py b/gdb/python/python-config.py
index 027d80d..4dbf6d2 100644
--- a/gdb/python/python-config.py
+++ b/gdb/python/python-config.py
@@ -3,6 +3,7 @@
 
 import sys
 import os
+import platform
 import getopt
 from distutils import sysconfig
 
@@ -49,10 +50,23 @@ def to_unix_path(path):
 
 for opt in opt_flags:
     if opt == "--prefix":
-        print(to_unix_path(sysconfig.PREFIX))
+        prefix=os.environ.get('CONFIG_PYTHON_PREFIX')
+        if prefix and prefix.strip():
+            sys.stderr.write ("%s -> [%s]\n" % (opt, prefix.strip()))
+            print (prefix.strip())
+        else:
+            sys.stderr.write ("%s -> [%s]\n" % (opt, to_unix_path(sysconfig.PREFIX)))
+            print (to_unix_path(sysconfig.PREFIX))
+ 
 
     elif opt == "--exec-prefix":
-        print(to_unix_path(sysconfig.EXEC_PREFIX))
+        prefix=os.environ.get('CONFIG_PYTHON_PREFIX')
+        if prefix and prefix.strip():
+            sys.stderr.write ("%s -> [%s]\n" % (opt, prefix.strip()))
+            print (prefix.strip())
+        else:
+            sys.stderr.write ("%s -> [%s]\n" % (opt, to_unix_path(sysconfig.EXEC_PREFIX)))
+            print (to_unix_path(sysconfig.EXEC_PREFIX))
 
     elif opt in ("--includes", "--cflags"):
         flags = [
@@ -61,6 +75,7 @@ for opt in opt_flags:
         ]
         if opt == "--cflags":
             flags.extend(getvar("CFLAGS").split())
+        sys.stderr.write ("%s -> [%s]\n" % (opt, to_unix_path(' '.join(flags))))
         print(to_unix_path(" ".join(flags)))
 
     elif opt in ("--libs", "--ldflags"):
@@ -79,4 +94,6 @@ for opt in opt_flags:
                     libs.insert(0, "-L" + sysconfig.PREFIX + "/libs")
             if getvar("LINKFORSHARED") is not None:
                 libs.extend(getvar("LINKFORSHARED").split())
-        print(to_unix_path(" ".join(libs)))
+        tmp = to_unix_path(' '.join(libs))
+        sys.stderr.write ("%s -> [%s]\n" % (opt, tmp))
+        print (tmp)
